patches/tModLoader/Terraria/IO/WorldFileData.tML.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using Terraria.ModLoader;
+using Terraria.ModLoader.IO;
+
+namespace Terraria.IO;
+
+public partial class WorldFileData
+{
+	internal IList<string> usedMods; // log this?
```

Comment:
Only if you also do so for players. Remove comment and make a new issue/PR if desired.

Reply:
#3768 will be that pr.